migrations/sql/V2019.06.26.10.30__add_code_required_report_tables.sql

```diff
@@ -0,0 +1,154 @@
+
+);
+
+COMMENT ON TABLE mine_report is 'A code required report for a specific mine or permit on a mine.';
+ALTER TABLE mine_report OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report_submission (
+    mine_report_submission_id             serial                                  NOT NULL PRIMARY KEY,
+    mine_report_submission_guid           uuid                                    NOT NULL            ,
+    mine_report_id                        integer                                 NOT NULL            ,
+    received_date                         timestamp                                                   ,
+    mine_report_submission_status_code    varchar(3)                              NOT NULL            ,
+    active_ind                            boolean DEFAULT true                    NOT NULL            ,
+    create_user                           character varying(60)                   NOT NULL            ,
+    create_timestamp                      timestamp with time zone DEFAULT now()  NOT NULL            ,
+    update_user                           character varying(60)                   NOT NULL            ,
+    update_timestamp                      timestamp with time zone DEFAULT now()  NOT NULL            ,
+
+    FOREIGN KEY (mine_report_id) REFERENCES mine_report(mine_report_id) DEFERRABLE INITIALLY DEFERRED ,
+    FOREIGN KEY (mine_report_submission_status_code) REFERENCES mine_report_submission_status_code(mine_report_submission_status_code) DEFERRABLE INITIALLY DEFERRED
+);
+
+COMMENT ON TABLE mine_report_submission is 'A file submission for a code required report.';
+ALTER TABLE mine_report_submission OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report_document_xref
+(
+    mine_report_document_xref_guid uuid    DEFAULT gen_random_uuid() NOT NULL PRIMARY KEY,
+    mine_document_guid             uuid                              NOT NULL            ,
+    mine_report_submission_id      integer                           NOT NULL            ,
+
+    FOREIGN KEY (mine_document_guid) REFERENCES mine_document(mine_document_guid) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (mine_report_submission_id) REFERENCES mine_report_submission(mine_report_submission_id) DEFERRABLE INITIALLY DEFERRED
+);
+
+COMMENT ON TABLE mine_report_document_xref is 'Links a mine document to report submission.';
+ALTER TABLE mine_report_document_xref OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report_comment (
+    mine_report_comment_id           serial                                  NOT NULL PRIMARY KEY,
+    mine_report_comment_guid         uuid                                    NOT NULL            ,
+    mine_report_id                   integer                                 NOT NULL            ,
+    mine_report_submission_id        integer                                                     ,
+    comment                          varchar(300)                            NOT NULL            ,
```

Comment:
"comment" may be a reserved word in the database... should be changed to "something"_comment. I would suggest a name but I'm not sure if this is a mine_report comment or a report_submission_comment.... if it's both, what would be the generic name... also, if this is a mutually exclusive relationship, we may want a database constraint to allow a record to tie to either the mine_report OR the mine_report_submission, but not both. If that's the case.

Reply:
So the way this was thought up is that the comment will always link to at least a mine report. It can sometimes be linked to a mine_report submission and if it is linked to a submission than this comment should only show up on that submission and not the report.